mlflow/gluon.py

```diff
@@ -0,0 +1,184 @@
+import os
```

Comment:
@cosmincatalin I noticed that this module doesn't produce MLflow Models with the `pyfunc` flavor. Is it possible to add a `pyfunc` representation so that the model is more broadly compatible with MLflow's supported model serving / deployment tools?

For reference, see the `mlflow.keras` module's pyfunc wrapper implementation: https://github.com/mlflow/mlflow/blob/7fd12b4decbee396fc7ecd0d828a99f288cc9397/mlflow/keras.py#L281-L333.

Reply:
Right, I'll try to do that.

Reply:
The problem with adding a `pyfunc` for Gluon is that I cannot provided a needed additional argument, without modifying the `load_model` function in `pyfunc/__init__.py`, which I suspect you're not willing to easily accept.

Here is the code that would be needed in `gluon.py`:

```
class _GluonModelWrapper:
    def __init__(self, gluon_model):
        self.gluon_model = gluon_model

    def predict(self, ndarray):
        return self.gluon_model(ndarray)


def _load_pyfunc(path, ctx):
    """
    Load PyFunc implementation. Called by ``pyfunc.load_pyfunc``.

    :param path: Local filesystem path to the MLflow Model with the ``gluon`` flavor.
    :param ctx: Either CPU or GPU.
    """
    m = load_model(path, ctx)
    return _GluonModelWrapper(m)
```

You'll notice that I need to provide a context (CPU or GPU) when loading a model. To accomplish that I would need something similar to this in `pyfunc/__init__.py`:

```
def load_model(model_uri, suppress_warnings=True, **kwargs):
    ...
    return importlib.import_module(conf[MAIN])._load_pyfunc(data_path, **kwargs)
```

Otherwise, loading a Gluon model works fine like this:

```
model_loaded = mlflow.gluon.load_model(model_path, ctx.cpu())
```

Let me know your thought, @dbczumar

Reply:
@cosmincatalin Is it possible to have the `pyfunc` representation create and use a CPU context by default? This would remove the need for another argument to `_load_pyfunc`. 

Longer term, I think it would probably make sense to expand the `_load_pyfunc` interface to accept optional args/kwargs.

Reply:
I can do that, no problem.